src/AzureReadOnlyStore.cs

```diff
@@ -97,27 +88,21 @@ public bool IsSameStore(IReadOnlyStore other)
         {
             var blobPrefix = $"{Realm}/";
 
-            var token = default(BlobContinuationToken);
-
-            var result = await Persistent.ListBlobsSegmentedAsync(
-                prefix: blobPrefix,
-                useFlatBlobListing: true,
-                blobListingDetails: BlobListingDetails.Metadata,
-                maxResults: null,
-                currentToken: token,
-                options: null,
-                operationContext: null,
-                cancellationToken: cancel).ConfigureAwait(false);
+            var result = await Persistent.GetBlobsAsync(
+                BlobTraits.Metadata,
+                BlobStates.None,
+                blobPrefix,
+                cancel).ToListAsync();
 
-            if (result.ContinuationToken != null) return false;
+            if (!result.Any()) return false;
 
-            foreach (var item in result.Results)
+            foreach (var item in result)
             {
-                if (!(item is CloudBlob blob)) continue;
+                if (!(item is BlobItem blob)) continue;
                 if (!Hash.TryParse(blob.Name.Substring(blob.Name.Length - 32), out var hash)) continue;
                 if (!(blob.Properties.LastModified is DateTimeOffset dto)) continue;
 
-                callback(hash, blob.Properties.Length, dto.UtcDateTime);
+                callback(hash, blob.Properties.ContentLength.GetValueOrDefault(), dto.UtcDateTime);
```

Comment:
If the value of `ContentLength` is not present, it is safer to throw than to continue with an incorrect default value.